### PR TITLE
Add more informative error when :max-queued-requests is less than :max-requests and 8

### LINKED ISSUE #354: An exception is raised when :max-queued-requests is lower than :min-thread when creating a jetty server
I've just updated ring-jetty-adapter from 1.6.3 to 1.7.1 and previously I had :max-queued-requests set to 0 while :max-threads and :min-threads where both set to 64. This now raises an IllegalArgumentException for BlockingArrayQueue, changing :max-queued-requests to 64 stops the error from appearing.

Would it be possible to Raise an exception with a more useful error message, for example ":min-queued-requests must be greater than or equal to :min-threads"?
Or possibly the value of :max-queued-requests could be set to :min-threads if it is less than :min-threads.

### PR DESCRIPTION
**Pull request by Akeboshiwind**
If accepted should close #354.

### CODE REVIEW

**Comment by Akeboshiwind** [comment]
Yeah, it's the minimum value that `queue-capacity` can have in the threadpool buffer.
See here: https://github.com/ring-clojure/ring/blob/master/ring-jetty-adapter/src/ring/adapter/jetty.clj#L105

The error could probably be worded better but I didn't want to mention `queue-capacity` in the error because that's an internal variable name.

**Comment by weavejester** [comment]
I've been looking into the reasons behind the current queuepool code, and it looks like it's designed to mirror the code in Jetty. The reason that there is an 8 minimum capacity is because that's what the [QueuedThreadPool](https://git.eclipse.org/c/jetty/org.eclipse.jetty.project.git/tree/jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java#n100) implements by default.

Now that we know this, we can see the intent is to allow a minimum default queue size even if the number of threads is very low. However, if someone is explicitly passing the `:max-queued-requests` option, then this default no longer applies. Therefore we should have something like:

```clojure
queue-max-capacity (options :max-queued-requests Integer/MAX_VALUE)
queue-capacity     (-> min-threads (max 8) (min queue-max-capacity))
```

In other words:

1. The queue capacity should be no greater than the max-queued-requests
2. By default, the queue capacity should be the number of threads
3. If the number of threads is below 8, then by default the queue capacity should be 8

Rule 1 takes precedence over rules 2 and 3.

So I think we can actually do away with the exceptions (sorry for telling you to write them), now that the reasons for the underlying code are better understood.

**Comment by Akeboshiwind** [comment]
No problem, you're correct this solution is better. Thanks for taking a more detailed look into the issue :)

I'll close this PR as presumably you will make the one line change that's needed.
